classes/class-swsales-reports.php

```diff
@@ -42,141 +73,263 @@ public static function show_reports_page() { ?>
-								?>
-								<option value="<?php esc_attr_e( $sale->get_id() ); ?>" <?php echo( esc_html( $selected_modifier ) ); ?>>
-									<?php echo( esc_html( $sale->get_name() ) ); ?>
-								</option>
+						<div class="swsales_reports-filters">
+							<?php 
+
+							for ($i = 0; $i < 2; $i++) {
+								$class_modifier = ( $i === 0 ) ? 'left' : 'right';
+								$label_wording = ( $i === 0 ) ? 'Show reports for' : 'vs.';
+								
+								$selected =  $sales_to_show !== null && array_key_exists( $i, $sales_to_show )  ? $sales_to_show[$i] : null;
+							?>
+								<label for="swsales_select_report_<?php echo $class_modifier ?>"><?php esc_html_e( $label_wording, 'sitewide-sales' ); ?></label>
+								<select id="swsales_select_report_<?php echo $class_modifier ?>"  name="sitewide_sale[]">
+									<?php
+									if($i == 1) {
+									?>
+									<option selected value="0"><?php esc_html_e( '- Choose One -', 'sitewide-sales' ); ?></option>
+									<?php
+									}
+									foreach ( $all_sitewide_sales as $sitewide_sale_id ) {
+										$sale   = SWSales_Sitewide_Sale::get_sitewide_sale( $sitewide_sale_id );
+										$selected_modifier =  $selected != null && $sale === $selected ? 'selected="selected"' : '';
+									?>
+										<option value="<?php esc_attr_e( $sale->get_id() ); ?>" <?php echo( esc_html( $selected_modifier ) ); ?>>
+											<?php echo( esc_html( $sale->get_name() ) ); ?>
+										</option>
+										<?php
+									}
+									?>
+								</select>
 								<?php
 							}
-							?>
-						</select>
+						?>
+						</div>
 					</form>
-					<hr />
+					<script>
+					jQuery(document).ready(function($) {
+						$('#swsales_select_report_left, #swsales_select_report_right').on('change', evt => {
+							$changed = $(evt.target);
+							$changed.closest('form').submit();
+							});
+						});
+					</script>
 					<?php
 				} else { ?>
 					<div class="sitewide_sales_message sitewide_sales_alert"><?php printf(__( 'No Sitewide Sales found. <a href="%s">Create your first Sitewide Sale &raquo;</a>', 'sitewide-sales' ), admin_url( 'post-new.php?post_type=sitewide_sale' ) ); ?></div>
 					<?php
 				}
 
 				// Show report for sitewide sale if applicable.
-				if ( null !== $sale_to_show ) {
-					SWSales_Reports::show_report( $sale_to_show );
-				}
+				SWSales_Reports::show_report( $sales_to_show );
 			?>
 		</div> <!-- sitewide-sales_admin -->
 		<?php
 	}
 
 	/**
-	 * Show report content for a Sitewide Sale.
+	 * Show report content for selected Sitewide Sales objects.
 	 *
-	 * @param SWSales_Sitewide_Sale $sitewide_sale to show report for.
+	 * @param Array An array of SWSales_Sitewide_Sale objects to show report for.
+	 * @since TBD.
 	 */
-	public static function show_report( $sitewide_sale ) {
-		if ( ! is_a( $sitewide_sale, 'Sitewide_Sales\classes\SWSales_Sitewide_Sale' ) ) {
+	public static function show_report( $sitewide_sales ) {
+		// Bail if the array comes empty
```

Comment:
Is there any chance people or other code was calling this show_report button and passing in a single report?

If so, we can account for that by check if ! is_array and casting $sitewide_sales to an array if not.

But maybe not worth the extra code if we feel no one would have called this themselves.